marklogic-client-api/src/main/java/com/marklogic/client/dataservices/impl/InputEndpointImpl.java

```diff
@@ -0,0 +1,40 @@
+package com.marklogic.client.dataservices.impl;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.marklogic.client.DatabaseClient;
+import com.marklogic.client.dataservices.InputEndpoint;
+import com.marklogic.client.dataservices.impl.IOEndpointImpl;
+import com.marklogic.client.io.marker.JSONWriteHandle;
+
+public class InputEndpointImpl extends IOEndpointImpl implements InputEndpoint {
+	private static int DEFAULT_BATCH_SIZE = 100;
+	private InputCallerImpl caller;
+	private int batchSize;
+	
+	public InputEndpointImpl(DatabaseClient client, JSONWriteHandle apiDecl) {
+		this(client, new InputCallerImpl(apiDecl));
+	}
+	
+	private InputEndpointImpl(DatabaseClient client, InputCallerImpl caller) {
+		super(client, caller);
+		this.caller = caller;
+		
+		JsonNode apiDeclaration = caller.getApiDeclaration();
+		if(apiDeclaration.has("$bulk") && apiDeclaration.get("$bulk").isObject() && 
+				apiDeclaration.get("$bulk").has("inputBatchSize")) {
```

Comment:
Would an additional test on apiDeclaration.get("$bulk").get("inputBatchSize").isInt() ensure that getting the value as an int won't fail?